src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -109,6 +115,13 @@ protected virtual bool CanUpdateProperty([NotNull] ModelMetadata propertyMetadat
 
         private async Task<bool> CanValueBindAnyModelProperties(MutableObjectBinderContext context)
         {
+            // If there are no properties on the model, there is nothing to bind. We are here means this is not a top 
+            // level object. So we return false.
+            if (context.PropertyMetadata == null || context.PropertyMetadata.Count == 0)
```

Comment:
I considered it and didn't feel strongly. Tradeoffs here are extra copy vs making it like 5% easier for implementors :laughing: 

We're taking a hard look at all of this extensibility for beta7, so any changes we make will last until then.